spring-cloud-azure-feature-management/README.md

```diff
@@ -0,0 +1,129 @@
+
+### Supported properties
+
+Name | Description | Required | Default 
+---|---|---|---
+spring.cloud.azure.feature.management.fail-fast | Whether throw RuntimeException or not when exception occurs | No |  true
+
+## Consumption
+The simplest use case for feature flags is to do a conditional check for whether a feature is enabled to take different paths in code. The use cases grow when additional using spring-cloud-azure-feature-flag-web to manage web based features.
+
+### Feature Check
+The basic form of feature management is checking if a feature is enabled and then performing actions based on the result. This is done through the autowiring `FeatureManager` and calling it's `isEnabled` method.
+
+```
+@Autowired
+FeatureManager featureManager;
+
+if(featureManager.isEnabled("FeatureT")) {
+    // Do Something
+}
+```
+
+`FeatureManager` can also be accessed by `@Component` classes.
+
+### Controllers
+When using the Feature Management Web library you can require that a given feature  is enabled in order to execute. This can be done by using the `@FeatureOn` annotation.
+
+```
+@GetMapping("/featureT")
+@FeatureGate(feature = "FeatureT")
+@ResponseBody
+public String featureT() {
+    ...
+}
+```
+
+The `featureT` endpoint can only be accessed if "FeatureT" is enabled.
+
+### Disabled Action Handling
+When a controller is blocked because the feature it specifies is disabled, `IDisabledFeaturesHandler` will be invoked. By default, a HTTP 404 is returned. This can be overridden using implementing `IDisabledFeaturesHandler`.
+
+```
+@Component
+public class DisabledFeaturesHandler implements IDisabledFeaturesHandler{
+
+    @Override
+    public HttpServletResponse handleDisabledFeatures(HttpServletRequest request, HttpServletResponse response) {
+        ...
+        return response;
+    }
+
+}
+
+```
+
+### Routing
+Certain routes may expose application capabilites that are gated by features. These routes can redirected if a feature is disabled to another endpoint.
+
+```
+@GetMapping("/featureT")
+@FeatureGate(feature = "FeatureT" fallback= "/oldEndpoint")
+@ResponseBody
+public String featureT() {
+    ...
+}
+
+@GetMapping("/oldEndpoint")
+@ResponseBody
+public String oldEndpoint() {
+    ...
+}
+```
+
+## Implementing a Feature Filter
+Creating a feature filter provides a way to enable features bassed on criteria that you define. To implement a feature filter, the `FeatureFilter` interface must be implemented. `FeatureFilter` has a single method `evaluate`. When a feature specifies that it can be enabled for a feature filter, the `evaluate` method is called. method is called. If `evaluate` returns `true` it means the feature should be enabled.
+
+Feature filters are found by being defined as `@Component` where there name matches the expected filter defined in the configuration.
```

Comment:
You may want to provide a code snippet so readers can see exactly how the name maps, what parameters they should be expect, what annotations they need to add etc. all in one place.